frontend/apps/reader/modules/readertoc.lua

```diff
@@ -436,21 +447,48 @@ function ReaderToc:isChapterEnd(cur_pageno)
 end
 
 function ReaderToc:getChapterPagesLeft(pageno)
-    --if self:isChapterEnd(pageno) then return 0 end
-    local next_chapter = self:getNextChapter(pageno)
-    if next_chapter then
-        next_chapter = next_chapter - pageno - 1
+    if self.ui.document:hasHiddenFlows() then
+        local flow = self.ui.document:getPageFlow(pageno)
+        -- Count pages until new chapter
+        local pages_left = 0
+        local test_page = self.ui.document:getNextPage(pageno)
+        while test_page > 0 do
+            pages_left = pages_left + 1
+            if self:isChapterStart(test_page) then
+                return pages_left
+            end
+            test_page = self.ui.document:getNextPage(test_page)
+        end
```

Comment:
Is it ok to return nil if there's no chapterStart after current page?

Reply:
It should be, it's the current behaviour, and I have tested it, I believe.